packages/web-client/app/services/hub-authentication.ts

```diff
@@ -37,41 +37,66 @@ export default class HubAuthentication extends Service {
 
   @task *initializeTask(): TaskGenerator<void> {
     yield waitForProperty(this.layer2Network, 'isInitializing', false);
-    if (!this.authToken) return;
-    if (!this.layer2Network.isConnected) {
-      this.storage.removeItem('authToken');
-      return;
-    }
-    if (!this.authToken) return;
-    let isAuthValid = yield this.layer2Network.checkHubAuthenticationValid(
-      this.authToken
-    );
-    if (isAuthValid) {
+    if (yield this.hasValidAuthentication()) {
       this.isAuthenticated = true;
     } else {
-      this.storage.removeItem('authToken');
+      this.authToken = null;
     }
   }
 
+  async hasValidAuthentication() {
+    return Boolean(
+      this.layer2Network.isConnected &&
+        this.authToken &&
+        (await this.layer2Network.checkHubAuthenticationValid(this.authToken))
+    );
+  }
+
   get authToken(): string | null {
     return this.storage.getItem('authToken');
   }
 
   set authToken(val: string | null) {
     if (val) {
+      // we're not setting isAuthenticated to true here because
+      // it is possible to mistakenly set authToken to an arbitrary truthy value
+      // in an external consumer
```

Comment:
I wonder if we could use the [`storage` event](https://developer.mozilla.org/en-US/docs/Web/API/Window/storage_event) to react to direct manipulation of the token in `localStorage` 🤔

Reply:
I was thinking about this too. I think it probably is good to listen for clearing/removal of the token from local storage.